server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -88,19 +88,28 @@ public abstract class FieldMapper extends Mapper {
      * @param sourceKeepMode   mode for storing the field source in synthetic source mode
      * @param hasScript         whether a script is defined for the field
      * @param onScriptError     the behaviour for when the defined script fails at runtime
+     * @param currentFieldIsAMultiField whether current field is part of a multi-field definition
      */
     protected record BuilderParams(
         MultiFields multiFields,
         CopyTo copyTo,
         Optional<SourceKeepMode> sourceKeepMode,
         boolean hasScript,
-        OnScriptError onScriptError
+        OnScriptError onScriptError,
+        boolean currentFieldIsAMultiField
```

Comment:
I am not sure this is needed, the `MappingParserContext` available in the `TypeParser` has a `isWithinMultiField` helper. Can we access this directly rather than duplicating the logic?

Reply:
Yes, I think we can determine whether a field is within mulit field from parser instead of determining that here. Let me change this.

Reply:
Something like this: ae04e6643e79925dff3e23e27e6cf7e78f2c35a4 ?

Reply:
Yep exactly.